### ISSUE TITLE
Constraint library support on client

### ISSUE THREAD

**Issue by Necrossin** [Facepunch/garrysmod-requests#2500]
### Details

I know this might be technically a duplicate (the only one I could find is from 2013 and that request ended up being unresolved), but still.

I wonder how hard would it be to implement a clientside support for `constraint` library on client, since that would be a huge gamechanger when making effects similar to `BoltImpact` and for other stuff that involve clientside ragdolls (welding bones together to have a ragdoll statue, etc).

**Comment by Palexite** [comment]
I agree with this. Otherwise in some instances we're forced to use physobj:ComputeShadowControl which is kind of messy + it does not constrain consistently as a weld would. It would allow gamemodes to have more workarounds in pursuing physics-based visual effects without the cost of the server.
